src/common/types/date_t.cpp

```diff
@@ -413,7 +413,7 @@ date_t Date::getLastDay(date_t& date) {
     return Date::fromDate(year, month, 1) - 1;
 }
 
-int32_t Date::getDatePart(DatePartSpecifier specifier, date_t& date) {
+int32_t Date::getDatePart(DatePartSpecifier specifier, date_t date) {
```

Comment:
What about its caller `Date::trunc`? Should that take in a reference of `date_t` after this change?

Can you double check if we have the same problem in other functions? I found we use references a lot in date related functions. I would carefully go through each of them to do some due intelligence checks.

Also, if we have to use reference to avoid copying overheads, I would suggest we should add `const` qualifier to make sure the input is safe without any changes.